content/300-guides/500-other/900-advanced-database-tasks/06-typesafe-raw-sql/01-gis-support.mdx

```diff
@@ -0,0 +1,112 @@
+---
+title: 'Using SafeQL for unsupported features'
+metaTitle: 'Using SafeQL for unsupported features (Guide)'
+metaDescription: 'Learn how to use SafeQL and Prisma Client extensions to add support for different queries.'
+---
+
+## Overview
+
+This page explains how to use Prisma Client extensions and SafeSQL in order to add support for features not natively supported by Prisma.
+
+Our example will be using [PostGIS](https://postgis.net/) but is applicable to other features and database engines.
+
+## Prerequisites
+
+To follow this guide, we recommend:
+
+- an existing [PostgreSQL](https://www.postgresql.org/) database
+- an existing Prisma project
+
+## 1. Set up Prisma for use with PostGIS
+
+If you haven't already, update your Prisma schema to add the `postgis` PostgreSQL extension
+
+```prisma
+generator client {
+  provider        = "prisma-client-js"
+  previewFeatures = ["postgresqlExtensions"]
+}
+
+datasource db {
+  provider   = "postgresql"
+  url        = env("DATABASE_URL")
+  extensions = [postgis]
+}
+```
+
+Then, create a migration using `npx prisma migrate dev` in order to enable the extension. The migration file should look like the following:
+
+```sql
+
+-- CreateExtension
+CREATE EXTENSION IF NOT EXISTS "postgis";
+
+```
+
+Make sure that the migration is applied!
+
+## 2. Create a new model that uses a GIS column
+
+After the migration is applied, you can now add a new model that has a GIS column. For our example we'll be using the following `PointOfInterest` model.
+
+```prisma
+model PointOfInterest {
+  id       Int                                   @id @default(autoincrement())
+  name     String
+  location Unsupported("geography(Point, 4326)")
+}
+```
+
+You'll notice that `location` is marked as Unsupported. This means that we lose a lot of the benefits of Prisma when working with `PointOfInterest`. We'll be using [SafeQL](https://safeql.dev/) to fix this.
+
+Create a new migration with `npx prisma migrate dev` and apply it to add the `PointOfInterest` table to our database.
+
+## 3. Integrate SafeQL
+
+SafeQL is easily integrated with Prisma. Follow [SafeQL's integration guide](https://safeql.dev/compatibility/prisma.html) in order to lint `$queryRaw` and `$executeRaw` Prisma calls.
```

Comment:
```suggestion
SafeQL is easily integrated with Prisma. Follow [SafeQL's integration guide](https://safeql.dev/compatibility/prisma.html) in order to lint `$queryRaw` and `$executeRaw` Prisma calls.

<Admonition>

Note that for SafeQL to run it will need to have access to your database. We recommend passing the same environment variable or connection string that is used by Prisma.

</Admonition>
```